### PR TITLE
doc: update CHANGES for 4.1

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Add major changes since 4.1a1 release.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -33,6 +33,20 @@
 # New collective algorithms. All collective algorithms are listed in
   src/mpi/coll/coll_algorithms.txt
 
+# Removed hydra2.
+
+# Added in-repository wiki documentations.
```
Comment (raffenet) [inline CHANGES:38]:
```suggestion
# Added in-repository wiki documentation.
```

File: CHANGES [hunk 2]
```diff
@@ -33,6 +33,20 @@
...
+
+# Added stream workq to support optimizations for enqueue operations.
+
+# Better support for large count APIs by eliminating type conversion issues.
+
+# PMI refactor.
+
+# Hydra refactor.
```
Comment (raffenet) [inline CHANGES:46]: I'm not sure these are notable on their own. Maybe we can call out specific new capabilities?

File: CHANGES [hunk 3]
```diff
@@ -33,6 +33,20 @@
 # New collective algorithms. All collective algorithms are listed in
   src/mpi/coll/coll_algorithms.txt
 
+# Removed hydra2.
```
Comment (raffenet) [inline CHANGES:36]: We should add something here about Hydra2 features being ported to Hydra.

File: CHANGES (hunk 1 above)
Comment (hzhou) [reply CHANGES]: fixed.

File: CHANGES (hunk 3 above)
Comment (hzhou) [reply CHANGES]: Good suggestion! done.

**Review by raffenet — APPROVED** [review]

**Pull request merged**